pkg/reconciler/apis/apibinding/apibinding_reconcile.go

```diff
@@ -483,32 +453,43 @@ func crdFromAPIResourceSchema(schema *apisv1alpha1.APIResourceSchema) (*apiexten
 	return crd, nil
 }
 
-// lockedStringSet guards a sets.String with an RWMutex.
-type lockedStringSet struct {
-	lock sync.RWMutex
-	s    sets.String
+func getAPIExportClusterName(apiBinding *apisv1alpha1.APIBinding) (logicalcluster.LogicalCluster, error) {
+	parent, hasParent := logicalcluster.From(apiBinding).Parent()
+	if !hasParent {
+		return logicalcluster.LogicalCluster{}, fmt.Errorf("a APIBinding in %s cannot reference a workspace", logicalcluster.From(apiBinding))
+	}
+	if apiBinding.Spec.Reference.Workspace == nil {
+		return logicalcluster.LogicalCluster{}, fmt.Errorf("APIBinding does not specify an APIExport")
+	}
+	return parent.Join(apiBinding.Spec.Reference.Workspace.WorkspaceName), nil
 }
 
-func newLockedStringSet(s ...string) *lockedStringSet {
-	return &lockedStringSet{
-		s: sets.NewString(s...),
+func referencedAPIExportChanged(apiBinding *apisv1alpha1.APIBinding) bool {
+	// Can't happen because of OpenAPI, but just in case
+	if apiBinding.Spec.Reference.Workspace == nil {
+		return false
 	}
-}
 
-func (l *lockedStringSet) Add(s string) {
-	l.lock.Lock()
-	defer l.lock.Unlock()
-	l.s.Insert(s)
-}
+	if apiBinding.Status.BoundAPIExport == nil {
+		return false
```

Comment:
had expected true here and below. So this means that "changed" is actually "was set + set to something else now". Might be clear to return true here and have the distinction between first-bind and rebind at the call site.

Reply:
How about just removing the 2 if blocks that check Status.BoundAPIExport?